staging/src/k8s.io/apimachinery/pkg/util/net/interface.go

```diff
@@ -52,36 +81,81 @@ func getRoutes(input io.Reader) ([]Route, error) {
 			continue
 		}
 		fields := strings.Fields(line)
-		routes = append(routes, Route{})
-		route := &routes[len(routes)-1]
-		route.Interface = fields[0]
-		ip, err := parseIP(fields[1])
+		dest, err := parseHexToIPv4(fields[1], familyIPv4)
+		if err != nil {
+			return nil, err
+		}
+		gw, err := parseHexToIPv4(fields[2], familyIPv4)
+		if err != nil {
+			return nil, err
+		}
+		if !dest.Equal(net.IPv4zero) {
+			continue
+		}
+		routes = append(routes, Route{
+			Interface:   fields[0],
+			Destination: dest,
+			Gateway:     gw,
+			Family:      familyIPv4,
+		})
+	}
+	return routes, nil
+}
+
+func getIPv6DefaultRoutes(input io.Reader) ([]Route, error) {
+	routes := []Route{}
+	scanner := bufio.NewReader(input)
+	for {
+		line, err := scanner.ReadString('\n')
+		if err == io.EOF {
+			break
+		}
+		fields := strings.Fields(line)
+		dest, err := parseHexToIPv4(fields[0], familyIPv6)
 		if err != nil {
 			return nil, err
 		}
-		route.Destination = ip
-		ip, err = parseIP(fields[2])
+		gw, err := parseHexToIPv4(fields[4], familyIPv6)
 		if err != nil {
 			return nil, err
 		}
-		route.Gateway = ip
+		if !dest.Equal(net.IPv6zero) {
+			continue
+		}
+		if gw.Equal(net.IPv6zero) {
+			continue // loopback
+		}
+		routes = append(routes, Route{
+			Interface:   fields[9],
+			Destination: dest,
+			Gateway:     gw,
+			Family:      familyIPv6,
+		})
 	}
 	return routes, nil
 }
 
-func parseIP(str string) (net.IP, error) {
+// parseHexToIPv4 takes the hex IP address string from route file and converts it
+// from little endian to big endian for creation of a net.IP address.
+// a net.IP, using big endian ordering.
+func parseHexToIPv4(str string, family AddressFamily) (net.IP, error) {
```

Comment:
Whoops, this should not have been renamed... this final patchset uses the method for both V4 and V6. Reverting back.